tests/bluechi_test/systemd_lists.py

```diff
@@ -0,0 +1,247 @@
+#
+# Copyright Contributors to the Eclipse BlueChi project
+#
+# SPDX-License-Identifier: LGPL-2.1-or-later
+
+import re
+from typing import AnyStr, Dict
+
+# Literals used during bluechictl/systemctl output parsing
+_ATTR_NODE_NAME = "node_name"
+_ATTR_UNIT_NAME = "unit_name"
+_ATTR_UNIT_FILE = "unit_file"
+_ATTR_PRESET = "preset"
+_ATTR_STATE = "state"
+_ATTR_SUB_STATE = "sub_state"
+_ATTR_LOADED = "loaded"
+_ATTR_DESCRIPTION = "description"
+
+
+# Pattern to parse bluechictl list-units line
+BC_LIST_UNITS_LINE_PAT = re.compile(
+    rf"""\s*(?P<{_ATTR_NODE_NAME}>[\S]+)\s*\|
+        \s*(?P<{_ATTR_UNIT_NAME}>[\S]+)\s*\|
+        \s*(?P<{_ATTR_STATE}>[\S]+)\s*\|
+        \s*(?P<{_ATTR_SUB_STATE}>[\S]+)\s*""",
+    re.VERBOSE,
+)
+
+
+# Pattern to parse bluechictl list-unit-files line
+BC_LIST_UNIT_FILES_LINE_PAT = re.compile(
+    rf"""\s*(?P<{_ATTR_NODE_NAME}>[\S]+)\s*\|
+        \s*((?:[^/]*/)*)(?P<{_ATTR_UNIT_FILE}>[\S]+)\s*\|
+        \s*(?P<{_ATTR_STATE}>[\S]+)\s*""",
+    re.VERBOSE,
+)
+
+
+# Pattern to parse systemctl list-units line
+SC_LIST_UNITS_LINE_PAT = re.compile(
+    rf"""\s*(?P<{_ATTR_UNIT_NAME}>\S+)
+        \s+(?P<{_ATTR_LOADED}>\S+)
+        \s+(?P<{_ATTR_STATE}>\S+)
+        \s+(?P<{_ATTR_SUB_STATE}>\S+)
+        \s+(?P<{_ATTR_DESCRIPTION}>.*)$""",
+    re.VERBOSE,
+)
+
+
+# Pattern to parse systemctl list-unit-files line
+SC_LIST_UNIT_FILES_LINE_PAT = re.compile(
+    rf"""\s*(?P<{_ATTR_UNIT_FILE}>\S+)
+        \s+(?P<{_ATTR_STATE}>\S+)
+        \s+(?P<{_ATTR_PRESET}>\S+)""",
+    re.VERBOSE,
+)
+
+
+# Pattern to remove escape sequences appearing on some systemctl output lines
+_SC_LINE_ESC_SEQ = re.compile(r"\x1B(?:[@-Z\\-_]|\[[0-?]*[ -/]*[@-~])")
```

Comment:
NIT: We could bundle these regex pattern into a dedicated class so that its easier to import and use:
```python
class RegexPattern {
   SYSTEMCTL_LIST_UNITS = ...
   BLUECHICTL_LIST_UNITS = ...
   ...
}
# ...

RegexPattern.SYSTEMCTL_LIST_UNITS
```

Reply:
Done